presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1366,13 +1376,8 @@ private Optional<TableHandle> processTableVersion(Table table, QualifiedObjectNa
                 }
             }
 
-            // Two block entries for table version type and expression.
-            BlockBuilder blockBuilder = asOfExprType.createBlockBuilder(null, 2);
-            writeNativeValue(asOfExprType, blockBuilder, tableVersionType.ordinal());
-            writeNativeValue(asOfExprType, blockBuilder, evalAsOfExpr);
-            Block block = blockBuilder.build();
-
-            return metadata.getHandleVersion(session, name, Optional.of(block));
+            ConnectorTableVersion tableVersion = new ConnectorTableVersion(toVersionType(tableVersionType), asOfExprType, evalAsOfExpr);
```

Comment:
Can we move toVersionType() logic to ConnectorTableVersion class instead? This would help avoid checking for and specifying enum values at two places.

Reply:
enum values from sql.tree package are not accessible from SPI or Iceberg modules, so I have used enum in SPI connector.